File: Problems/C/383.c
int cmp(const void *a, const void *b)
{
    return *(const char*)a - *(const char*)b;
}

bool canConstruct(char * ransomNote, char * magazine){
    int ransomNoteSize = strlen(ransomNote);
    int magazineSize = strlen(magazine);
    int i = 0, j = 0;
    
    qsort(ransomNote, ransomNoteSize, sizeof(char), cmp);
    qsort(magazine, magazineSize, sizeof(char), cmp);
    
    while((i != ransomNoteSize)&&(j != magazineSize))
    {
        if(ransomNote[i] == magazine[j])
        {
            i++;
            j++;
        }
        else if(ransomNote[i] < magazine[j])
            return false;
        else
            j++;
    }
    
    return i == ransomNoteSize;
}

File: Problems/Python 3/383.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        magazineCount = Counter(magazine)
        ransomNoteCount = Counter(ransomNote)
        
        for c in ransomNoteCount:
            if c not in ransomNoteCount or magazineCount[c] < ransomNoteCount[c]:
                return False
            
        return True
